pkg/exporter/openapi3_test.go

```diff
@@ -0,0 +1,165 @@
+	errorPropertiesMessage := gjson.Get(string(outputSpecJSON), "components.schemas.Error.properties.message.type")
+	assert.Equal(t, "object", errorType.Str)
+	assert.Equal(t, "integer", errorPropertiesCode.Str)
+	assert.Equal(t, "int64", errorPropertiesCodeFormat.Str)
+	assert.Equal(t, "string", errorPropertiesMessage.Str)
+
+	getPetsSuccessResponse := gjson.Get(string(outputSpecJSON), "paths./pets.get.responses.200.content.application/json.schema.$ref")
+	getPetsErrorResponse := gjson.Get(string(outputSpecJSON), "paths./pets.get.responses.default.content.application/json.schema.$ref")
+	postPetsRequired := gjson.Get(string(outputSpecJSON), "paths./pets.post.requestBody.required")
+	assert.Equal(t, "#/components/schemas/NewPetResponse", getPetsSuccessResponse.Str)
+	assert.Equal(t, "#/components/schemas/Error", getPetsErrorResponse.Str)
+	assert.True(t, postPetsRequired.Bool())
+}
+func TestExport(t *testing.T) {
+	type1 := syslwrapper.MakeMap(
+		syslwrapper.MakeTypeRef("app1", []string{"login"}, "app2", []string{"request"}),
+		syslwrapper.MakePrimitive("string"))
+	type2 := syslwrapper.MakePrimitive("string")
+	param1 := syslwrapper.MakeParam("Login", type1)
+	var app1 = syslwrapper.MakeApp("app1", []*sysl.Param{param1}, map[string]*sysl.Type{})
+	var app2 = syslwrapper.MakeApp("app2", []*sysl.Param{}, map[string]*sysl.Type{"request": type2})
+	var mod = &sysl.Module{
+		Apps: map[string]*sysl.Application{
+			"app1": &app1,
+			"app2": &app2,
+		},
+	}
+
+	mapper := syslwrapper.MakeAppMapper(mod)
+	mapper.IndexTypes()
+	simpleApps, err := mapper.Map()
+	assert.NoError(t, err)
+
+	exporter := MakeOpenAPI3Exporter(simpleApps, &logrus.Logger{})
+	err = exporter.Export()
+	assert.NoError(t, err)
+
+	outputSpec, err := exporter.SerializeOutput("app1", "json")
+	assert.NoError(t, err)
+
+	paramName := gjson.Get(string(outputSpec), "paths.testEndpoint.get.parameters.0.name")
+	assert.Equal(t, "Login", paramName.Str)
+	paramSchema := gjson.Get(string(outputSpec), "paths.testEndpoint.get.parameters.0.schema.type")
+	assert.Equal(t, "object", paramSchema.Str)
+}
+
+func TestMapOptional(t *testing.T) {
+	simpleApps := map[string]*syslwrapper.App{
+		"TestApp": {
+			Name: "TestApp",
+			Endpoints: map[string]*syslwrapper.Endpoint{
+				"TestEndpoint": {
+					Summary: "GetPets",
+					Path:    "GET /pets",
+					Params: map[string]*syslwrapper.Parameter{
+						"limit": {
+							In:   "query",
+							Name: "limit",
+							Type: &syslwrapper.Type{
+								Type:     "int",
+								Optional: true,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	exporter := MakeOpenAPI3Exporter(simpleApps, &logrus.Logger{})
+	err := exporter.Export()
+	assert.NoError(t, err)
+	limitRequired := exporter.openapi3["TestApp"].Paths["/pets"].Get.Parameters.GetByInAndName("query", "limit").Required
+	assert.False(t, limitRequired)
+}
+
+func TestMapEnums(t *testing.T) {
+	simpleApps := map[string]*syslwrapper.App{
+		"TestApp": {
```

Comment:
I like the idea of this, but I think it might be a little cleaner if you could just make an app/module directly from a string; currently you need to pass a file system in

Reply:
Sounds good, will make a utility function for that in a future PR